src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -1948,15 +1949,76 @@ std::vector<MergeTreeMutationStatus> ReplicatedMergeTreeQueue::getMutationsStatu
     return result;
 }
 
+namespace
+{
+
+struct BlockInfoInZooKeeper
+{
+    String partition;
+    Int64 number;
+    String zk_path;
+    std::future<Coordination::GetResponse> contents_future;
+};
+
+template <bool with_multiread>
+std::vector<BlockInfoInZooKeeper> getBlockInfos(const auto & partitions, const auto & zookeeper, const auto & zookeeper_path)
```

Comment:
There are much more places where we send multiple get requests asynchronously. Consider implementing such method in ZooKeeper client, that will take parent path and array (or iterators range) of node names, check API version and retrieve data using async get requests or MultiRead

Reply:
I've been thinking about it, but didn't do it as part of this task because it could end up a bit complex.
Mostly because of the way how we process with async gets, when we get the result of the next get request we instantly process it, we don't wait for all of them to finish.

One way to implement it would be to send a callback to be called on each get request but I can do it as part of a different PR.